libicyque.c

```diff
@@ -168,6 +177,26 @@ icq_get_url_sign(IcyQueAccount *ia, gboolean is_post, const gchar *url, const gc
     return ret;
 }
 
+static JsonObject*
+icq_generate_robusto_request(IcyQueAccount *ia, const gchar* method, JsonObject* params)
+{
+	JsonObject *robustoRequest = json_object_new();
+	
+	json_object_set_string_member(robustoRequest, "authToken", ia->robusto_token);
+	json_object_set_string_member(robustoRequest, "method", method);
+	
+	GString *request_id = g_string_new(NULL);
+	g_string_append_printf(request_id, "%lu-%ld", ia->robusto_request_id++, time(NULL) - ia->server_time_offset);
+	json_object_set_string_member(robustoRequest, "reqId", request_id->str);
+	
+	if(ia->robusto_client_id >= 0) {
+		json_object_set_int_member(robustoRequest, "clientId", ia->robusto_client_id);
+	}
+	
+	json_object_set_object_member(robustoRequest, "params", params);
+	return robustoRequest;

```

Comment:
You'll need to `g_string_free` the request_id to prevent a memleak

Reply:
Does `json_object_set_string_member(..., request_id->str)` copy the string into an internally managed string, so that a `g_string_free` directly after that line is okay?

Reply:
That's correct